data/mods/ssb/scripts.ts

```diff
@@ -100,324 +212,650 @@ export const Scripts: ModdedBattleScriptsData = {
+			this.runEvent('AfterMoveSecondary', targets, pokemon, move);
 		}
-		const item = pokemon.getItem();
-		if (!item.zMove) return;
-		if (item.itemUser && !item.itemUser.includes(pokemon.species.name)) return;
-		let atLeastOne = false;
-		let mustStruggle = true;
-		const zMoves: ZMoveOptions = [];
-		for (const moveSlot of pokemon.moveSlots) {
-			if (moveSlot.pp <= 0) {
-				zMoves.push(null);
-				continue;
-			}
-			if (!moveSlot.disabled) {
-				mustStruggle = false;
-			}
-			const move = this.dex.getMove(moveSlot.move);
-			let zMoveName = this.getZMove(move, pokemon, true) || '';
-			if (zMoveName) {
-				const zMove = this.dex.getMove(zMoveName);
-				if (!zMove.isZ && zMove.category === 'Status') zMoveName = "Z-" + zMoveName;
-				zMoves.push({move: zMoveName, target: zMove.target});
+		return undefined;
+	},
+	hitStepMoveHitLoop(targets, pokemon, move) { // Temporary name
+		const damage: (number | boolean | undefined)[] = [];
+		for (const i of targets.keys()) {
+			damage[i] = 0;
+		}
+		move.totalDamage = 0;
+		pokemon.lastDamage = 0;
+		let targetHits = move.multihit || 1;
+		if (Array.isArray(targetHits)) {
+			// yes, it's hardcoded... meh
+			if (targetHits[0] === 2 && targetHits[1] === 5) {
+				if (this.gen >= 5) {
+					targetHits = this.sample([2, 2, 3, 3, 4, 5]);
+				} else {
+					targetHits = this.sample([2, 2, 2, 3, 3, 3, 4, 5]);
+				}
 			} else {
-				zMoves.push(null);
+				targetHits = this.random(targetHits[0], targetHits[1] + 1);
 			}
-			if (zMoveName) atLeastOne = true;
 		}
-		if (atLeastOne && !mustStruggle) return zMoves;
-	},
-	runZPower(move, pokemon) {
-		const zPower = this.dex.getEffect('zpower');
-		if (move.category !== 'Status') {
-			this.attrLastMove('[zeffect]');
-		} else if (move.zMove?.boost) {
-			this.boost(move.zMove.boost, pokemon, pokemon, zPower);
-		} else {
-			switch (move.zMove?.effect) {
-			case 'heal':
-				this.heal(pokemon.maxhp, pokemon, pokemon, zPower);
-				break;
-			case 'healhalf':
-				// For DragonWhale
-				this.heal(pokemon.baseMaxhp / 2, pokemon, pokemon, zPower);
-				break;
-			case 'healreplacement':
-				move.self = {sideCondition: 'healreplacement'};
-				break;
-			case 'boostreplacement':
-				// For nui
-				move.self = {sideCondition: 'boostreplacement'};
-				break;
-			case 'clearnegativeboost':
-				const boosts: SparseBoostsTable = {};
-				let i: BoostName;
-				for (i in pokemon.boosts) {
-					if (pokemon.boosts[i] < 0) {
-						boosts[i] = 0;
+		targetHits = Math.floor(targetHits);
+		let nullDamage = true;
+		let moveDamage: (number | boolean | undefined)[];
+		// There is no need to recursively check the ´sleepUsable´ flag as Sleep Talk can only be used while asleep.
+		const isSleepUsable = move.sleepUsable || this.dex.getMove(move.sourceEffect).sleepUsable;
+
+		let targetsCopy: (Pokemon | false | null)[] = targets.slice(0);
+		let hit: number;
+		for (hit = 1; hit <= targetHits; hit++) {
```

Comment:
```suggestion
		for (let hit = 1; hit <= targetHits; hit++) {
```

Reply:
no, `hit` is used elsewhere in the function after iterating